python/ngen_cal/src/ngen/cal/ngen.py

```diff
@@ -46,13 +46,13 @@ def _params_as_df(params: Mapping[str, Parameters], name: str = None):
             df['model'] = k
             df.rename(columns={'name':'param'}, inplace=True)
             dfs.append(df)
-        return pd.concat(dfs)
+        return pd.concat(dfs).set_index('param')
     else:
         p = params.get(name, [])
         df = pd.DataFrame([s.__dict__ for s in p])
         df['model'] = name
         df.rename(columns={'name':'param'}, inplace=True)
-        return df
+        return df.set_index('param')
```

Comment:
Ugh.  It is actually possible for this dataframe to be empty if called from _map_params_to_realization and there is a module in the realization which doesn't have any params from the config to map to it, in this case, the `set_index` here will fail

Reply:
Also, I think this breaks search, since `param` is no longer a column when subset to update_config

search.py", line 168, in dds_set